packages/modules/order/src/services/order-module-service.ts

```diff
@@ -844,6 +851,45 @@ export default class OrderModuleService
     })
   }
 
+  @InjectTransactionManager()
+  // @ts-expect-error
+  async deleteOrders(
+    orderIds: string | string[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<void> {
+    const ids = Array.isArray(orderIds) ? orderIds : [orderIds]
+
+    const orders = await this.orderService_.list(
+      { id: ids },
+      {
+        select: ["id", "shipping_address_id", "billing_address_id"],
+      },
+      sharedContext
+    )
+
+    const orderAddressIds = orders
+      .map((order) => [order.shipping_address_id, order.billing_address_id])
+      .flat(1)
+
+    const orderChanges = await this.orderChangeService_.list(
+      { order_id: ids },
+      { select: ["id"] },
+      sharedContext
+    )
+
+    const orderChangeIds = orderChanges.map((orderChange) => orderChange.id)
+
+    await Promise.all([
+      this.orderAddressService_.delete(orderAddressIds, sharedContext),
+      this.orderChangeService_.delete(orderChangeIds, sharedContext),
+    ])
+
+    // Delete order changes & actions
+    await this.orderChangeService_.delete(orderChangeIds, sharedContext)
+
+    await super.deleteOrders(ids, sharedContext)
```

Comment:
We can add a couple more entities to be deleted here:
Follow the pseudo code of the tables. You can just use the internal services to delete them.

```
select item_id from order_item where order_id in (ids)

delete from "order_line_item" where id in ( [result of previous query] )

delete order_item where order_id in (ids)


select shipping_method_id from order_shipping where order_id in (ids)

delete from order_shipping_method where order_id in ( [result of previous query] )

delete from order_shipping where order_id in (ids)

delete order_summary where order_id in (ids)
delete from order_transaction where order_id in ()
```

These will delete in cascade item and shipping adjustments and tax_lines.